### PR TITLE
add new tag startupMode

### PR DESCRIPTION
**Pull request by viktoriyabogdanova**
Notes:
* DTSPO-15113 - add new tag startupMode
*
*

### CODE REVIEW

File: vars/spinInfra.groovy [hunk 1]
```diff
@@ -68,6 +68,9 @@ def call(Map<String, ?> params) {
         if (Environment.toTagName(config.environment) != "production") {
           tags = tags + [autoShutdown: "true"]
         }
+        if (Environment.toTagName(config.environment) != "production") {
+          tags = tags + [startupMode: "always"]
```
Comment (timja) [inline vars/spinInfra.groovy:72]: shouldn't this be adding onDemand to sandbox? and what are you doing about old resources is there going to be a script ran to backfill or will it default to always? (even then a script would be needed for the likes of ITHC wouldn't it for when we enable it there?)

Comment (viktoriyabogdanova) [reply vars/spinInfra.groovy:72]: Weren't we supposed to make it default always and then overwrite it per env per resource?

Comment (timja) [reply vars/spinInfra.groovy:72]: it should be set at an environment level I think.

Comment (timja) [reply vars/spinInfra.groovy:72]: cc @adusumillipraveen

Comment (adusumillipraveen) [reply vars/spinInfra.groovy:72]: yeah, let's have a chat @viktoriyabogdanova , i think `always` is kind of default if not added. We need the other onDemand one for sandbox/ithc. And yes, we may have to do a script anyway to add tags to existing resources when we roll this out

**Review by timja — APPROVED** [review]

**Pull request merged**